mne/forward/_make_forward.py

```diff
@@ -218,86 +226,145 @@ def _setup_bem(bem, bem_extra, neeg, mri_head_t, verbose=None):
+        ncomp = 0
+
+    _print_coord_trans(info['dev_head_t'])
 
-    if neeg <= 0 and nmeg <= 0:
-        raise RuntimeError('Could not find any MEG or EEG channels')
+    # Make info structure to allow making compensator later
+    ncomp_data = len(info['comps'])
+    ref_meg = True if not ignore_ref else False
+    picks = pick_types(info, meg=True, ref_meg=ref_meg, exclude=exclude)
+    meg_info = pick_info(info, picks) if nmeg > 0 else None
 
     # Create coil descriptions with transformation to head or MRI frame
-    templates = _read_coil_defs()
-    if nmeg > 0 and ncomp > 0:  # Compensation channel information
-        logger.info('%d compensation data sets in %s'
-                    % (ncomp_data, info_extra))
+    templates = _read_coil_defs(elekta_defs=elekta_defs)
     if nmeg > 0:
-        megcoils = _create_coils(megchs, accuracy,
-                                 info['dev_head_t'], 'meg', templates)
+        megcoils = _create_coils(megchs, accuracy, info['dev_head_t'], 'meg',
+                                 templates)
         if ncomp > 0:
+            logger.info('%d compensation data sets in %s' % (ncomp_data,
+                                                             info_extra))
             compcoils = _create_coils(compchs, FIFF.FWD_COIL_ACCURACY_NORMAL,
                                       info['dev_head_t'], 'meg', templates)
+        logger.info('Head coordinate MEG coil definitions created.')
+    else:
+        raise RuntimeError('Could not find any MEG channels')
+
+    return megcoils, compcoils, megnames, meg_info
+
+
+@verbose
+def _prep_eeg_channels(info, exclude='bads', verbose=None):
+    """Prepare EEG electrode definitions for forward calculation
+
+    Parameters
+    ----------
+    info : instance of Info
+        The measurement information dictionary
+    exclude : list of str | str
+        List of channels to exclude. If 'bads' (default), exclude channels in
+        info['bads']
+    verbose : bool, str, int, or None
+        If not None, override default verbose level (see mne.verbose).
+        Defaults to raw.verbose.
+
+    Returns
+    -------
+    eegels : list of dict
+        Information for each prepped EEG electrode
+    eegnames : list of str
+        Name of each prepped EEG electrode
+    """
+    eegnames, eegels = [], []
+    info_extra = 'info'
+
+    # Find EEG electrodes
+    picks = pick_types(info, meg=False, eeg=True, ref_meg=False,
+                       exclude=exclude)
+    neeg = len(picks)
+
+    # Get channel info and names for EEG channels
+    if neeg > 0:
+        eegchs = pick_info(info, picks)['chs']
+        eegnames = [info['ch_names'][p] for p in picks]
+        logger.info('Read %3d EEG channels from %s' % (len(picks), info_extra))
+
+    # Create EEG electrode descriptions
+    templates = _read_coil_defs()
     if neeg > 0:
         eegels = _create_coils(eegchs, coil_type='eeg', coilset=templates)
```

Comment:
No reason to have this under a separate conditional, can be combined with the one directly above it